### PR TITLE
Include ListComplete method in mock interfaces

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Update comment for client constructors.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes:
https://github.com/Azure/azure-sdk-for-go/issues/6559
https://github.com/Azure/azure-sdk-for-go/issues/6605

File: src/Templates/ServiceClientTemplate.cshtml [hunk 1]
```diff
@@ -67,7 +67,7 @@ else
 {
 <text>
 @EmptyLine
-@WrapComment("// ", string.Format("NewWithBaseURI creates an instance of the {0} client.", Model.BaseClient))
+@WrapComment("// ", string.Format("NewWithBaseURI creates an instance of the {0} client using a custom endpoint.  Use this when interacting with an Azure cloud that uses a non-standard base URI (sovereign clouds, Azure stack).", Model.BaseClient))
```
Comment (catalinaperalta) [inline src/Templates/ServiceClientTemplate.cshtml]: Just checking, we don't want the name of the client in this method name? (like: New{0}WithBaseURI)

Comment (jhendrixMSFT) [reply src/Templates/ServiceClientTemplate.cshtml]: In track 1 we have these base clients that don't have a specific name, see https://github.com/Azure/azure-sdk-for-go/blob/master/services/redis/mgmt/2018-03-01/redis/client.go#L45 for an example.

Comment (catalinaperalta) [reply src/Templates/ServiceClientTemplate.cshtml]: I see, thanks for the info!

**Comment by jhendrixMSFT** [comment]
CC @ArcturusZhang

**Comment by fearthecowboy** [comment]
# 🤖 AutoRest automatic publish job 🤖
## success (version: 2.1.140)

**Pull request merged**